### PR TITLE
[DH-145] fix pagination, basically a complete rewrite

### PR DESCRIPTION
**Pull request by shaneknapp**

### CODE REVIEW

File: scripts/delete-unused-users.py [hunk 1]
```diff
@@ -10,60 +10,138 @@
...
 log in again the next time they use the hub, but that's probably
 ok.
+
+Core functionality from @minrk:
+https://discourse.jupyter.org/t/is-there-a-way-to-bulk-delete-old-users/20866/3
 """
 import argparse
-from jhub_client.api import JupyterHubAPI
-from dateutil.parser import parse
 import asyncio
```
Comment (ryanlovett) [inline scripts/delete-unused-users.py]: I think `asyncio` would no longer be needed in the rewrite.

File: scripts/delete-unused-users.py [hunk 2]
```diff
@@ -10,60 +10,138 @@
...
 from datetime import timedelta, datetime
+from dateutil.parser import parse
+from jhub_client.api import JupyterHubAPI
```
Comment (ryanlovett) [inline scripts/delete-unused-users.py]: Not critical: per [PEP8](https://peps.python.org/pep-0008/#imports), third party imports should follow those from standard library.

**Review by ryanlovett — APPROVED** [review]
Looks good, but I haven't run it. :)

**Comment by shaneknapp** [comment]
> Looks good, but I haven't run it. :)

usernames redacted, but here's a sample run from the a11y hub:
```
(dh) ➜  scripts git:(fix-pagination-bug) ✗ ./delete-unused-users.py --hub_url https://a11y.datahub.berkeley.edu/
1: deleting XXXX
2: deleting XXXX
3: deleting XXXX
4: deleting XXXX
5: deleting XXXX
6: deleting XXXX
7: deleting XXXX
8: deleting XXXX
9: deleting XXXX
Deleted 9 total users.
```

--debug and --verbose also work as expected :)

**Pull request merged**